<launch>

    <arg name="type" default="gazebo" /> <!-- gazebo, jmavsim, none (only clover packages) -->
    <arg name="gui" default="true" /> <!-- Run Gazebo with GUI -->
    
    <!-- Gazebo instance -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch" if="$(eval type == 'gazebo')">
        <!-- Workaround for crashes in VMware -->
        <env name="SVGA_VGPU10" value="0" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="world_name" value="$(find clover_simulation)/resources/worlds/clover_aruco.world" />
        <arg name="verbose" value="true" />
    </include>

    <group ns="clover0">
        
        <include file="$(find swarm_in_blocks)/launch/single_drone.launch">
            <arg name="mav_id" value="0"/>
            <arg name="x" value="0.0"/>
            <arg name="y" value="0.0"/>
            <arg name="launch_gazebo" default="false" /> 
        </include>
            
    </group>

    <group ns="clover1">

        <include file="$(find swarm_in_blocks)/launch/single_drone.launch">
            <arg name="mav_id" value="1"/>
            <arg name="x" value="1.0"/>
            <arg name="y" value="0.0"/>
            <arg name="launch_gazebo" default="false" /> 
        </include>

    </group>

    <group ns="clover2">

        <include file="$(find swarm_in_blocks)/launch/single_drone.launch">
            <arg name="mav_id" value="2"/>
            <arg name="x" value="0.0"/>
            <arg name="y" value="1.0"/>
            <arg name="launch_gazebo" default="false" /> 
        </include>

    </group>
    
</launch>
